### PR TITLE
feature/create-chat

### PR DESCRIPTION
**Pull request by vladimir-maklyakov**

### CODE REVIEW

File: app/controllers/ChatController.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+let ChatService = require('../services/ChatService');
+
+class ChatController {
+  static get prefix() {
+    return '/chat';
+  }
+
+  static findById(req, res) {
+    ChatService.findById(req.params.id)
+      .then(chat => res.status(200).json(new ChatDto(chat)))
```
Comment (nikita-usatenko) [inline app/controllers/ChatController.js]: Service should return DTO object

File: app/controllers/ChatController.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      .catch(() => res.sendStatus(500));
+  }
+
+  static create(req, res) {
+    ChatService.create(req.body.users)
+      .then(chat => res.status(200).json(new ChatDto(chat)))
```
Comment (nikita-usatenko) [inline app/controllers/ChatController.js]: Same here

File: app/controllers/UserController.js [hunk 3]
```diff
@@ -1,19 +1,15 @@
...
   static index(req, res) {
-    User.find({}, (err, users) => {
-      if (err) {
-        res.sendStatus(500);
-        return;
-      }
-      res.status(200).json(users.map(user => new UserDto(user)));
-    });
+    UserService.findAll()
+      .then(response => res.status(200).json(response.map(user => new UserDto(user))))
```
Comment (nikita-usatenko) [inline app/controllers/UserController.js]: Same here

File: app/controllers/ChatController.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  static get prefix() {
+    return '/chat';
+  }
+
+  static findById(req, res) {
+    ChatService.findById(req.params.id)
+      .then(chat => res.status(200).json(chat))
+      .catch(() => res.sendStatus(500));
+  }
+
```
Comment (nikita-usatenko) [inline app/controllers/ChatController.js]: Get all chats for current user.

File: app/services/ChatService.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+let Chat = require('../models/Chat');
+let ChatDto = require('../dtos/ChatDto');
+
+class ChatService {
+  static findById(id) {
```
Comment (nikita-usatenko) [inline app/services/ChatService.js]: Service should receive DTO as parameter

File: app/controllers/ChatController.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  static create(req, res) {
+    ChatService.create(req.body.users)
```
Comment (nikita-usatenko) [inline app/controllers/ChatController.js]: Service should receive DTO as parameters

File: app/services/ChatService.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    return new Promise((resolve, reject) => Chat.findOne({ _id: id }, (err, chat) => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(new ChatDto(chat));
+    }));
+  }
+
+  static create(users) {
```
Comment (nikita-usatenko) [inline app/services/ChatService.js]: Rename users to userDtos so that it is more obvious?

File: app/controllers/ChatController.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+let ChatService = require('../services/ChatService');
+
+class ChatController {
+  static get prefix() {
+    return '/chat';
+  }
+
+  static findById(req, res) {
+    ChatService.findById(req.params.id)
```
Comment (nikita-usatenko) [inline app/controllers/ChatController.js]: Service should receive DTO as parameters

File: app/models/Chat.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+let mongoose = require('mongoose');
+
+const ChatSchema = new mongoose.Schema({
+  users: [String],
```
Comment (nikita-usatenko) [inline app/models/Chat.js]: Schema.Types.ObjectId

**Comment by vladimir-maklyakov** [comment]
[Create Chat - Server](https://trello.com/c/OyKawTsQ/65-create-chat-server)

**Comment by vladimir-maklyakov** [comment]
[Create Chat - Server](https://trello.com/c/OyKawTsQ/65-create-chat-server)

**Pull request merged**